### PR TITLE
Catalog Ingestion Product Changes Observer

### PR DESCRIPTION
**Pull request by garmashdev**
# Description
Observer class implemented for "catalog_product_save_commit_after" event to detect product changes and publish bolt product event instance for sending in future to bolt.

https://app.asana.com/0/1201931884901947/1201906341917644/f

#changelog Catalog Ingestion Product Changes Observer

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 

- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my ticket link and provided a changelog message.

### CODE REVIEW

**Comment by garmashdev** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1559** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1559" target="_blank">Graphite</a> 
    * **PR #1560** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1560" target="_blank">Graphite</a> 
      * **PR #1561** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1561" target="_blank">Graphite</a> 
        * **PR #1562** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1562" target="_blank">Graphite</a>  👈
          * **PR #1563** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1563" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1562?utm_source=stack-comment).

**Review by vitaliyreznikov — COMMENTED** [review]
Looks good, please add tests

File: Observer/PublishBoltProductEventObserver.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * Process event on 'save_commit_after' event
+     *
+     * @param Observer $observer
+     * @return void
+     */
+    public function execute(Observer $observer)
+    {
+        /** @var Product $product */
+        $product = $observer->getEvent()->getProduct();
```
Comment (vitaliyreznikov) [inline Observer/PublishBoltProductEventObserver.php:55]: Let's check the feature switch before this line

**Review by vitaliyreznikov — APPROVED** [review]
LGTM, one small comment

**Pull request merged**